native/common/jp_context.cpp

```diff
@@ -171,14 +199,16 @@ void JPContext::startJVM(const string& vmPath, const StringVector& args,
 		// is initialized.  Any other frame creation will result in an error.
 		JPJavaFrame frame(this, env);
 
-		JPException_init(frame);
+		jclass cls = (jclass) frame.FindClass("java/lang/Throwable");
+		m_Throwable_GetCauseID = frame.GetMethodID(cls, "getCause", "()Ljava/lang/Throwable;");
+
 		// After the JVM is created but before the context is started, we need
 		// to set up all the services that the context will need.
 		JP_TRACE("Initialize");
 
 		// We need these first because if anything goes south this is the first
 		// thing that will get hit.
-		jclass cls = frame.FindClass("java/lang/Object");
+		cls = frame.FindClass("java/lang/Object");
```

Comment:
Is this some kind of variable recycling?

Reply:
Yeah, cls gets used about 20 times here with each class.  I suppose I can rename it new names each time.   I just don't want any of these items to get cycled into globals like we had before.